src/main/java/spike/parser/Parser.java

```diff
@@ -0,0 +1,238 @@
+            }
+            LocalDateTime eventT = parseDateTime(command.substring(command.indexOf("/at") + 4), dtf);
+            if (!(eventT == null)) {
+                Event newE = new Event(command.substring(command.indexOf("event") + 6,
+                        command.indexOf("/at") - 1), eventT);
+                return new AddCommand(newE);
+            } else {
+                throw new SpikeException("Please enter a valid date in the format yyyy-MM-dd HHmm");
+            }
+        default:
+            return new IncorrectCommand("Should not reach this line");
+        }
+    }
+
+    private Command parseDelete(String[] commandWords, TaskList tasks) throws SpikeException {
+        if (commandWords.length != 2 || isInt(commandWords[1]) == -1
+                || isInt(commandWords[1]) > tasks.getListSize()) {
+            throw new SpikeException("Invalid arguments for deletion. Please check again!");
+        }
+        Task toDelete = tasks.getTasks().get(Integer.parseInt(commandWords[1]) - 1);
+        return new DeleteCommand(toDelete);
+    }
+
+    private Command parseToggleMark(CommandName c, String[] commandWords, TaskList tasks) throws SpikeException {
+        if (c.getCommand().equals("mark")) {
+            if (commandWords.length != 2 || isInt(commandWords[1]) == -1
+                    || isInt(commandWords[1]) > tasks.getListSize()) {
+                throw new SpikeException("Invalid arguments for marking. Please check again!");
+            }
+            Task toMark = tasks.getTasks().get(Integer.parseInt(commandWords[1]) - 1);
+            return new ToggleMarkCommand(1, toMark);
+        } else {
+            if (commandWords.length != 2 || isInt(commandWords[1]) == -1
+                    || isInt(commandWords[1]) > tasks.getListSize()) {
+                throw new SpikeException("Invalid arguments for unmarking. Please check again!");
+            }
+            Task toUnmark = tasks.getTasks().get(Integer.parseInt(commandWords[1]) - 1);
+            return new ToggleMarkCommand(0, toUnmark);
+        }
+    }
+
+    private Command parseExit() {
+        return new ExitCommand();
+    }
+
+
+    /**
+     * Parses date and time input by user and returns valid LocalDateTime object
+     */
+    private static LocalDateTime parseDateTime(String s, DateTimeFormatter dtf) {
+        try {
+            LocalDateTime dateTime = LocalDateTime.parse(s, dtf);
+            return dateTime;
+        } catch (DateTimeParseException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Checks whether the input string is integer.
+     * If yes, return it, else return -1.
+     */
+    private static int isInt(String str) {
+        try {
+            int x = Integer.parseInt(str);
+            return x; // it is an integer
+        } catch (NumberFormatException e) {
+            return -1; // not an integer
+        }
+
+    }
```

Comment:
Incomplete Javadocs comments, `@return` is missing (same as others which are incomplete)